services/common/rs_server_common/stac_api_common.py

```diff
@@ -0,0 +1,619 @@
+        queryables = {}
+
+        if self.adgs:
+            queryables["product:type"] = QueryableField(
+                type="string",
+                title="productType",
+                format="string",
+                description="String",
+            )
+
+        # If the collection has the Satellite field hard-coded with a single value,
+        # the user cannot query on platform and constellation
+        can_query = True
+        if collection_id:
+            satellites = self.select_config(collection_id).get("query", {}).get("Satellite", "")
+            if "," not in satellites:
+                can_query = False
+
+        # Read all platforms and constellations from the configuration file
+        if can_query:
+            config = {}
+            for satellite in map_stac_platform().get("satellites", {}):
+                config.update(satellite)
+            platforms = sorted(set(config.keys()))
+            connstellations = sorted(
+                set([platform["constellation"] for platform in config.values() if "constellation" in platform]),
+            )
+            queryables.update(
+                {
+                    "platform": QueryableField(
+                        type="string",
+                        title="platform",
+                        format="string",
+                        description="String",
+                        enum=platforms,
+                    ),
+                    "constellation": QueryableField(
+                        type="string",
+                        title="constellation",
+                        format="string",
+                        description="String",
+                        enum=connstellations,
+                    ),
+                },
+            )
+
+        return queryables
+
+    async def fetchval(self, query, *args, column=0, timeout=None):
+        """Run a query and return a value in the first row.
+
+        :param str query: Query text.
+        :param args: Query arguments.
+        :param int column: Numeric index within the record of the value to
+                           return (defaults to 0).
+        :param float timeout: Optional timeout value in seconds.
+                            If not specified, defaults to the value of
+                            ``command_timeout`` argument to the ``Connection``
+                            instance constructor.
+
+        :return: The value of the specified column of the first record, or
+                 None if no records were returned by the query.
+        """
+        query = query.strip()
+
+        # From stac_fastapi.pgstac.core.CoreCrudClient::all_collections
+        if query == "SELECT * FROM all_collections();":
+            return filter_allowed_collections(self.all_collections(), self.service, self.request)
+
+        # From stac_fastapi.pgstac.core.CoreCrudClient::get_collection
+        if query == "SELECT * FROM get_collection($1::text);":
```

Comment:
This looks like a limitation of pgstac (or whatever we use), we either use ORM or SQL, mixing them it's a confusion factor

Reply:
This code is taken from `stac_fastapi.pgstac.core.CoreCrudClient::get_collection`, in fact my class has to mock these queries made by stac_fastapi so I cannot change this query.